Source/WebCore/css/CSSPrimitiveValue.cpp

```diff
@@ -615,52 +615,47 @@ Ref<CSSPrimitiveValue> CSSPrimitiveValue::createURI(String value)
     return adoptRef(*new CSSPrimitiveValue(WTFMove(value), CSSUnitType::CSS_URI));
 }
 
-double CSSPrimitiveValue::computeDegrees() const
+template<> int CSSPrimitiveValue::resolveAsLength(const CSSToLengthConversionData& conversionData) const
 {
-    return computeDegrees(primitiveType(), doubleValue());
+    return roundForImpreciseConversion<int>(resolveAsLengthDouble(conversionData));
 }
 
-template<> int CSSPrimitiveValue::computeLength(const CSSToLengthConversionData& conversionData) const
+template<> unsigned CSSPrimitiveValue::resolveAsLength(const CSSToLengthConversionData& conversionData) const
 {
-    return roundForImpreciseConversion<int>(computeLengthDouble(conversionData));
+    return roundForImpreciseConversion<unsigned>(resolveAsLengthDouble(conversionData));
 }
 
-template<> unsigned CSSPrimitiveValue::computeLength(const CSSToLengthConversionData& conversionData) const
+template<> Length CSSPrimitiveValue::resolveAsLength(const CSSToLengthConversionData& conversionData) const
 {
-    return roundForImpreciseConversion<unsigned>(computeLengthDouble(conversionData));
+    return Length(clampTo<double>(resolveAsLengthDouble(conversionData), minValueForCssLength, maxValueForCssLength), LengthType::Fixed);
 }
 
-template<> Length CSSPrimitiveValue::computeLength(const CSSToLengthConversionData& conversionData) const
+template<> short CSSPrimitiveValue::resolveAsLength(const CSSToLengthConversionData& conversionData) const
 {
-    return Length(clampTo<double>(computeLengthDouble(conversionData), minValueForCssLength, maxValueForCssLength), LengthType::Fixed);
+    return roundForImpreciseConversion<short>(resolveAsLengthDouble(conversionData));
 }
 
-template<> short CSSPrimitiveValue::computeLength(const CSSToLengthConversionData& conversionData) const
+template<> unsigned short CSSPrimitiveValue::resolveAsLength(const CSSToLengthConversionData& conversionData) const
 {
-    return roundForImpreciseConversion<short>(computeLengthDouble(conversionData));
+    return roundForImpreciseConversion<unsigned short>(resolveAsLengthDouble(conversionData));
 }
 
-template<> unsigned short CSSPrimitiveValue::computeLength(const CSSToLengthConversionData& conversionData) const
+template<> float CSSPrimitiveValue::resolveAsLength(const CSSToLengthConversionData& conversionData) const
 {
-    return roundForImpreciseConversion<unsigned short>(computeLengthDouble(conversionData));
+    return static_cast<float>(resolveAsLengthDouble(conversionData));
```

Comment:
Does the `static_cast` here do anything? What would happen with no cast at all? Do we still have `narrowPrecisionToFloat` and when would we use that instead?

Reply:
Went with `narrowPrecisionToFloat`.